controllers/backup_controller.go

```diff
@@ -142,7 +142,7 @@ func (r *BackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		tryFlagBackupAsFailed(ctx, r.Client, &backup, fmt.Errorf("while getting pod: %w", err))
 		r.Recorder.Eventf(&backup, "Warning", "FindingPod", "Error getting target pod: %s",
 			cluster.Status.TargetPrimary)
-		return ctrl.Result{}, r.Status().Update(ctx, &backup)
+		return ctrl.Result{}, nil
```

Comment:
I think you miss the `r.Status().Update(ctx, &backup)` call here

Reply:
tryFlagBackupAsFailed does that